custom_components/waste_collection_schedule/config_flow.py

```diff
@@ -0,0 +1,384 @@
+        )
+        title = module.TITLE
+        errors = {}
+        description_placeholders = {}
+        # If all args are filled in
+        if args_input is not None:
+            # if contains method:
+            (
+                errors,
+                description_placeholders,
+                options,
+            ) = await self.__validate_args_user_input(self._source, args_input, module)
+            if len(errors) == 0:
+                return self.async_create_entry(
+                    title=title,
+                    data={CONF_SOURCE_NAME: self._source, CONF_SOURCE_ARGS: args_input},
+                    options=options,
+                )
+        return self.async_show_form(
+            step_id="args",
+            data_schema=schema,
+            errors=errors,
+            description_placeholders=description_placeholders,
+        )
+
+    def async_get_options_flow(self):
+        return WasteCollectionOptionsFlow(self)
+
+    async def async_step_reconfigure(self, args_input: dict[str, Any] | None = None):
+        config_entry = self.hass.config_entries.async_get_entry(
+            self.context["entry_id"]
+        )
+        if config_entry is None:
+            return self.async_abort(reason="reconfigure_failed")
+
+        source = config_entry.data["name"]
+        schema, module = await self.__get_arg_schema(
+            source, config_entry.data["args"], args_input, include_title=False
+        )
+        title = module.TITLE
+        errors: dict[str, str] = {}
+        description_placeholders: dict[str, str] = {}
+        # If all args are filled in
+        if args_input is not None:
+            # if contains method:
+            (
+                errors,
+                description_placeholders,
+                options,
+            ) = await self.__validate_args_user_input(source, args_input, module)
+            if len(errors) == 0:
+                data = {**config_entry.data}
+                data.update({CONF_SOURCE_NAME: source, CONF_SOURCE_ARGS: args_input})
+                _LOGGER.debug("reconfigured_data:")
+                _LOGGER.debug(data)
+                return self.async_update_reload_and_abort(
+                    config_entry,
+                    title=title,
+                    unique_id=config_entry.unique_id,
+                    data=data,
+                    options=options,
+                    reason="reconfigure_successful",
+                )
+        return self.async_show_form(
+            step_id="reconfigure",
+            data_schema=schema,
+            errors=errors,
+            description_placeholders=description_placeholders,
+        )
+
+
+class WasteCollectionOptionsFlow(OptionsFlow):
+    """Options flow."""
+
+    def __init__(self, entry) -> None:
+        self._entry = entry
+
+    async def async_step_init(self, options):
+        SCHEMA = vol.Schema(
+            {
+                vol.Optional(CONF_SOURCE_CALENDAR_TITLE): cv.string,
```

Comment:
The actual/default value is missing here. Test case: Create a calendar with a custom title. Open the options dialog. Result: custom name is missing in the dialog.